### PR TITLE
Added env options for Parent connections

### PR DESCRIPTION
**Pull request by TheBurchLog**
Added env options to allow for docker-compose deployment of children. That way it is easier to stand up example instances for users.

### CODE REVIEW

File: src/app/beer_garden/config.py [hunk 1]
```diff
@@ -771,6 +771,7 @@ def _parse_args(args: Sequence[str]) -> Tuple[YapconfSpec, dict]:
                             "type": "bool",
                             "default": False,
                             "description": "Publish events to parent garden over HTTP",
+                            "alt_env_names": ["PARENT_HTTP"],
```
[the remark this answers is not in the document]
Comment (TheBurchLog) [reply src/app/beer_garden/config.py:774]: Do we need to use Alt Env Names for when we are going to be passing those variables in through the command line with a different name?

[the remark this answers is not in the document]
Comment (hazmat345) [reply src/app/beer_garden/config.py:774]: @TheBurchLog no, the alt_env_names only affect how that config is resolved from the environment. If you're worried about the CLI name for this being overly long (`--event-parent-http-enable`) we could consider moving this section out from the events section - I'm not sure it really belongs there anyway.

[the remark this answers is not in the document]
Comment (TheBurchLog) [reply src/app/beer_garden/config.py:774]: sorry, I misspoke in my response. I need those to resolve from the environment variables. That is how I am defining it in the Docker Compose file. That way it was easier for me to initialize Beer Garden.

I agree that looking at the config it is not intuitive for the Parent connect information to be under events. We should probably pull that up to a top level config. I will look at moving that up.

File: src/app/beer_garden/config.py [hunk 2]
```diff
@@ -723,6 +723,95 @@ def _parse_args(args: Sequence[str]) -> Tuple[YapconfSpec, dict]:
...
+                    "alt_env_names": ["PARENT_HTTP"],
+                },
+                "ssl": {
+                    "type": "dict",
+                    "items": {
+                        "enabled": {
+                            "type": "bool",
+                            "default": False,
+                            "description": "Serve content using SSL",
+                            "cli_separator": "_",
```
Comment (hazmat345) [inline src/app/beer_garden/config.py:745]: Do we need this to be `_`?

File: src/app/beer_garden/config.py [hunk 3]
```diff
@@ -723,6 +723,95 @@ def _parse_args(args: Sequence[str]) -> Tuple[YapconfSpec, dict]:
...
+    "type": "dict",
+    "items": {
+        "http": {
+            "type": "dict",
+            "items": {
+                "enable": {
+                    "type": "bool",
+                    "default": False,
+                    "description": "Publish events to parent garden over HTTP",
+                    "alt_env_names": ["PARENT_HTTP"],
```
Comment (hazmat345) [inline src/app/beer_garden/config.py:736]: I *guess* I can be ok with this. I'd prefer to just set the normal name `PARENT_HTTP_ENABLE` over adding an alt name that is the same as the parent_http config item. But if you think it makes a difference and it works then :shrug:

File: src/app/beer_garden/config.py [hunk 4]
```diff
@@ -723,6 +723,95 @@ def _parse_args(args: Sequence[str]) -> Tuple[YapconfSpec, dict]:
...
+                    "type": "str",
+                    "default": "/",
+                    "description": "URL path prefix",
+                    "required": False,
+                },
+                "host": {
+                    "type": "str",
+                    "default": "0.0.0.0",
+                    "description": "Host for the HTTP Server to bind to",
+                    "alt_env_names": ["PARENT_HOST"],
```
Comment (hazmat345) [inline src/app/beer_garden/config.py:796]: This alt name should be removed - it's setting us up for problems whenever we add additional transport mechanisms. If you need to set this from compose just use the normal name `PARENT_HTTP_HOST`.

File: src/app/beer_garden/config.py [hunk 5]
```diff
@@ -733,7 +733,7 @@ def _parse_args(args: Sequence[str]) -> Tuple[YapconfSpec, dict]:
                     "type": "bool",
                     "default": False,
                     "description": "Publish events to parent garden over HTTP",
-                    "alt_env_names": ["PARENT_HTTP"],
+                    "alt_env_names": ["PARENT_HTTP_ENABLE"],
```
Comment (hazmat345) [inline src/app/beer_garden/config.py]: We don't need to specify this here - "PARENT_HTTP_ENABLE" *is* the environment name.

File: src/app/beer_garden/config.py [hunk 6]
```diff
@@ -793,7 +793,7 @@ def _parse_args(args: Sequence[str]) -> Tuple[YapconfSpec, dict]:
                     "type": "str",
                     "default": "0.0.0.0",
                     "description": "Host for the HTTP Server to bind to",
-                    "alt_env_names": ["PARENT_HOST"],
+                    "alt_env_names": ["PARENT_HTTP_HOST"],
```
Comment (hazmat345) [inline src/app/beer_garden/config.py]: Same with this - "PARENT_HTTP_HOST" is already the environment name for this config item.

File: docker/docker-compose/docker-compose.yml [hunk 7]
```diff
@@ -10,6 +10,13 @@ x-definitions: &env
     # Should be resolvable from inside and outside the docker network
     BG_PUBLISH_HOSTNAME: localhost
 
+    # IMPORTANT!!
+    # If you are connection to a Parent BG instance, toggle true
+    BG_PARENT_HTTP: false
+    
+    # IMPORTANT !!
+    # If you are connecting to a Parent BG instance, change this the resolvable DNS entry
+    BG_PARENT_HOST: localhost
```
Comment (hazmat345) [inline docker/docker-compose/docker-compose.yml:19]: These need to change - I'll change them.

**Pull request merged**